server_http.hpp

```diff
@@ -217,7 +217,7 @@ namespace SimpleWeb {
         
         ///Use this function if you need to recursively send parts of a longer message
         void send(const std::shared_ptr<Response> &response, const std::function<void(const boost::system::error_code&)>& callback=nullptr) const {
-            boost::asio::async_write(*response->socket, response->streambuf, [this, response, callback](const boost::system::error_code& ec, size_t /*bytes_transferred*/) {
```

Comment:
Yes, but it has to be done somewhere since Boost.Asio is still quite low-level and does not support smart pointer parameters.